code/template/filter/abstract.php

```diff
@@ -96,6 +96,47 @@ public function parseAttributes($string)
         return $result;
     }
 
+    /**
+     * Build an HTML element
+     *
+     * @param string $tag HTML tag name
+     * @param array  $attributes Key/Value pairs for the attributes
+     * @param string $children Child elements, not applicable for self-closing tags
+     * @return string
+     *
+     * Example:
+     * ```php
+     * echo $this->buildElement('a', ['href' => 'https://example.com/'], 'example link');
+     * // returns '<a href="https://example.com/">example link</a>
+     *
+     * echo $this->buildElement('meta', ['name' => 'foo', 'content' => 'bar']);
+     * // returns '<meta name="foo" content="bar" />
+     *
+     * ```
+     */
+    public function buildElement($tag, $attributes = [], $children = '')
+    {
+        static $self_closing_tags = [
+            'area', 'base', 'br', 'col', 'embed', 'hr', 'img', 'input', 'link',
+            'meta', 'param', 'source', 'track', 'wbr',
+            'ktml:script', 'ktml:style', 'ktml:include', 'ktml:content', 'ktml:toolbar', 'ktml:wrapper',
```

Comment:
@ercanozkaya Do we need this list of self closing tags? Can we just assume that if $content is null (which is the default) the tag self-closes? If content is not null, for example an empty string it would be a normal closing tag.

Also I would make $content accept both a string, and a callback. This would allow to easily return dynamic content.

Reply:
@johanjanssens I made $content accept callables as well. Good idea. 

About the list, there are many cases where we create empty elements for JS to fill in later. These are usually spans and divs. We would have to pass an empty string as the third parameter for these which would look very unusual.

Reply:
Oki, fair, point. I'm just having a beef with 'ktml:script', 'ktml:style', 'ktml:include', 'ktml:content', 'ktml:toolbar', 'ktml:wrapper'. These are custom and if you add new ones you need to specify them, and the list is hardcoded. 

Another approach could be that if the closing tag is not defined, we fallback on  the rule explained above. This gives you flexibility.